client/ayon_silhouette/api/plugin.py

```diff
@@ -124,11 +144,11 @@ def create(self, product_name, instance_data, pre_create_data):
 
     def collect_instances(self):
         shared_data = cache_instance_data(self.collection_shared_data)
-        for obj in shared_data["silhouette_cached_instances"].get(
+        for obj, uuid, data in shared_data["silhouette_cached_instances"].get(
                 self.identifier, []):
-
-            data = lib.read(obj)
-            data["instance_id"] = str(obj.id)
+            node_id = str(obj.id)
```

Comment:
Is this happening only because you don't have access to transient data?

Reply:
If yes, can you wait for this https://github.com/ynput/ayon-core/pull/1110 to be merged?

BTW why to handle it with changing `instance_id` instead of storing `node_id` to instance data?

Reply:
> BTW why to handle it with changing instance_id instead of storing node_id to instance data? Because currently you have to know that instance_id can hold "instance id" and "node id" -> I don't understand why you have to change instance_id?

Basically I need a unique id for the instance, and the id is usually persisted into the scene (although technically it doesn't have to if it's built up completely from scene data to begin with).

Anyway, I'll refactor to that PR and see how far I can get.